Add Assisted Installer metrics to metrics_allowlist

flaper87 opened the pull request:
Here's the list of name, types, and description for the assisted metrics being added in this PR

* assisted_installer_cluster_creations (counter): Number of cluster
  resources created, by version

* assisted_installer_cluster_installation_started (counter): Number of
  clusters that entered “installing” state, by version

* assisted_installer_cluster_installation_second (histogram):
  Histogram/sum/count of installation time for completed clusters, by
  result and OCP version (1 per cluster)

* assisted_installer_cluster_host_installation_count (histogram): Number
  of hosts per cluster when the installation started (1 per cluster)

* assisted_installer_host_installation_phase_seconds (histogram):
  Histogram/sum/count of time for each phase, by phase, final install
  result, and OCP version (Number of phases (~5) per Number of hosts in the cluster)

* assisted_installer_cluster_host_disk_sync_duration_ms (histogram):
  Histogram/sum/count of the disk's fdatasync duration (fetched from fio) (1 per host)

* assisted_installer_cluster_host_image_pull_status (histogram):
  Histogram / sum / count of the images' pull statuses (1 per image)

* assisted_installer_filesystem_usage_percentage (Gauge): The percentage
  of the filesystem usage by the service


There could be >=1k clusters at any point in time, which may have more than 1 host each.

Signed-off-by: Flavio Percoco <flavio@redhat.com>

flaper87 commented:
/cc @bjoydeep since he's been guiding me through the entire process

flaper87 commented:
/test test-e2e

clyang82 commented:
Thanks @flaper87 for your PR.
Could you please append the number of timeseries for the above metrics per scrape interval? I want to know the performance impact. Thanks.

flaper87 commented:
> Thanks @flaper87 for your PR.
> Could you please append the number of timeseries for the above metrics per scrape interval? I want to know the performance impact. Thanks.

I have updated the description to include the number of timeseries collected and added a note on how many clusters, and hosts there could be.

Hope this helps with the review :)

clyang82 commented:
Yes. That helps. Thanks

/lgtm

flaper87 commented:
Thank you for having such a smooth and easy process! :pray:
